### DISCUSSION TITLE
New layout: Organogram

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#825] [Ideas]
An idea to achieve the Organogram diagram:

Searching for organogram in google images, I noticed that it is usually built like this:
-upper levels: horizontal
-Lower levels: vertical
![image](https://user-images.githubusercontent.com/77707706/201993285-7b7e5ebc-5b5e-4a0f-b6e4-89ddaa404a2f.png)
Examples:
![image](https://user-images.githubusercontent.com/77707706/201992850-13da4ab3-2a8c-4ba2-a833-cd85aa436040.png)

In FP, now it's possible to make this interleave of horizontal and vertical layout. The closest to the example above is having the lower levels with the layout "align first child with parent", which results in this:
![image](https://user-images.githubusercontent.com/77707706/201994453-bd416648-edcf-4475-bb03-6d640f62a6eb.png)

But, notice that the Senior level is not clear as in the top examples, because the first junior gets in the middle of each senior. 

So, it seems to me that, to achieve a standard organogram layout, it would be useful to have this kind of alignment where the first child is a bit below the parent (like the Outline view, in FP). 

I guess, other types of diagrams would also benefit from that. For example, in the fish bone diagram, the first level causes would be more clearly apart from second level causes.

**Comment by euu2021** [comment]
Also, it would be possible to get closer to the timeline layout (as shown [here](https://github.com/freeplane/freeplane/discussions/819#discussioncomment-4126161)), because the "years" could be all aligned horizontally, with nothing between them.

**Comment by cesar404** [comment]
I would suggest a new layout with the children edges originating from the bottom (or the top) of their parent node, as we can see in the images below:
![cognitive-psychology-teiwv-1247533768056](https://user-images.githubusercontent.com/2542431/202006815-3b8b63c3-559b-43d1-9e53-876a9cdcca65.jpg)
![alternative outline view 4](https://user-images.githubusercontent.com/2542431/202006851-ceaf6862-1efb-4408-8a40-c97cfbd3a8ce.JPG)
![example-mind-map-timeline](https://user-images.githubusercontent.com/2542431/202006859-5d4477b0-3064-4a18-be94-e04c97e26195.png)
![FAQ-example-mind-map-top-down-blue](https://user-images.githubusercontent.com/2542431/202006875-1f6fe117-57e9-49a4-bed6-1e4e6242e9e5.jpg)
![SAV2C86_Map1](https://user-images.githubusercontent.com/2542431/202069662-3bfd581b-203f-494d-9ff4-18bc62ee88d0.png)

This combination of layouts optimizes considerably the use of the space and prevent the mind map from spreading too wide. This new suggested layout looks like an outline view, but it is slightly different.

**Reply by abc16361** [reply]
If FP has this layout, it would be easy to create a kanban-style map.

**Reply by dpolivaev** [reply]
Nice, I am going to work into this direction

**Comment by cesar404** [comment]
Just some more images showing different instances of the proposed new layout:
![1200_800](https://user-images.githubusercontent.com/2542431/202871854-6fee0ddf-b73b-422b-a9d4-b1591c5d9a9b.jpg)
![SharePoint+IA](https://user-images.githubusercontent.com/2542431/202871873-3bdbeaad-72a8-448c-b317-1458130df0d6.png)
![1200_8002](https://user-images.githubusercontent.com/2542431/202871875-2627542e-b86b-4bf5-a2e4-9e26224a5f68.jpg)
![Sign Up _ LinkedIn](https://user-images.githubusercontent.com/2542431/202872183-9db5a7a3-ae4a-48b9-b657-2c6431a45aaa.png)
![243107385_4863956676947965_1145084268240687537_n](https://user-images.githubusercontent.com/2542431/202877808-486bcd94-eaf1-440e-ac71-32e8d95893b1.jpg)
![337-ecommerce-sitemap-template](https://user-images.githubusercontent.com/2542431/202879746-c8f28946-ff00-4823-b819-8805b76e7671.png)
![vertikale-organigramme](https://user-images.githubusercontent.com/2542431/202871878-3e241e10-ed7d-403d-bae4-2c3a57abdb2f.png)

**Comment by dpolivaev** [comment]
There is a new preview version 1.11.1_03 alpha with more compact layouts following suggestions from @cesar404.
Unfortunately, it is not fully compatible with the previous 1.11.1 previews.

Please check it out and share your feedback.

**Comment by cesar404** [comment]
Nice! 
@dpolivaev , I suspect there is a problem with the preview file with Java. The file size is only 1.3mb. I couldn't install it on Windows 10.

**Reply by dpolivaev** [reply]
Yes, thanks. I have got to my computer again and reuploaded it.

**Reply by cesar404** [reply]
![test](https://user-images.githubusercontent.com/2542431/204154923-209fc2b0-cde3-49bd-8b0a-8126974eb3b8.png)

Very nice!
I'll perform some tests and give you my feedback.

**Reply by cesar404** [reply]
Ok, I took a quick look at the new preview version and it works great, but I do have a few observations:

1) When I apply "horizontal layout" on a branch, the edge lines are overlapped , every single time, by the nodes, as you can see on the left branch in the picture below. I have to reposition the nodes manually so that the branch looks correct, as shown on the right branch in the picture below.
![horizontal layout overlapping](https://user-images.githubusercontent.com/2542431/204183370-73213799-c233-462d-a0bd-bae6d5400d58.png)

2) When I use horizontal edge and apply on a branch a "both sides" variant of the newest layout, there is not enough indentation between the parent edge line and the side (border) of the child nodes, as shown on the left branch in the picture below. It should look more like the branch on the right.
![both sides indentation](https://user-images.githubusercontent.com/2542431/204185691-c17ddc2c-8f1b-410b-82d3-bde903f93521.png)

3) Again, when I use  a "both sides" variant of the newest layout, the point of origin of the edge line looks ugly:
![origin](https://user-images.githubusercontent.com/2542431/204187462-bdb7ae08-ae86-4070-9cf7-02311a593cd2.png)

4) When I use horizontal edges and apply traditional "vertical layout" inside a "horizontal layout" branch, the points of origin of the children edges are different. When I use "smoothly curved" edge, they look alright.
![traditional vertical layout](https://user-images.githubusercontent.com/2542431/204189160-33789015-9f82-4a30-83c6-9aada02e9290.png)

5) It is not always clear what each layout icon really do exactly. For example: I have no idea what that gear means. Also the names of the layouts are long and confusing. I'm sure they can be simplified. The image below (taken from Mindmapper) shows more friendly icons, at least in my opinion:
![QZW2uvzHtI](https://user-images.githubusercontent.com/2542431/204191132-d14bf8c8-d63f-4951-a988-2b9f2dcd3def.png)

**Reply by cesar404** [reply]
6) When I use fork nodes on a "top-to-bottom both sides" layout in conjunction with horizontal edges, the edge line is not properly drawn (it's thinner than the edge width value). See the branch on the left in the image below: 
![both sides fork](https://user-images.githubusercontent.com/2542431/204690596-1ad32b62-3de9-41c2-a617-168a6cb4589c.png)
But when I use the bezier curve type of edge, this issue doesn't happen (see image below): 
![fork belzier both sides](https://user-images.githubusercontent.com/2542431/204690605-615b5887-6c22-4576-a27f-40c1e165a244.png)

**Reply by dpolivaev** [reply]
@cesar404  I have tried to fix the bugs you reported in the new preview version 1.11.1_04 currently available at https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

I agree that icons and names for the layouts could be improved, but it is a different issue and I am going to start a separate discussion about it.

**Comment by abc16361** [comment]
Just did a simple test. It's pretty good!
![kanban](https://user-images.githubusercontent.com/101362122/204159156-3c6cf1bb-284f-45a9-a908-31a4a824c83c.png)

**Comment by guoranjun2** [comment]
I just test new layout and found that timeline layout can be implemented now, that's great.

![image](https://user-images.githubusercontent.com/50692082/204442295-0fc2fc81-4719-4e5d-babd-a062e26314c0.png)

**Reply by guoranjun2** [reply]
the only question is that the node would be covered by others in some situation when you combile different layout.
![image](https://user-images.githubusercontent.com/50692082/204442730-eb418784-d316-4ad1-be60-36b8202c9ed4.png)
and it supposed to be like this:
![image](https://user-images.githubusercontent.com/50692082/204442827-eade5606-d9d7-4839-85a4-70a82f162afd.png)

**Reply by dpolivaev** [reply]
If the layout followed your suggestion, the node indicated by the arrow would have to change its position relative to its parent node each time it is folded or unfolded. And if its position on the screen were kept, it would make all other nodes change their on-screen positions. Therefore new Frepelane layouts intentionally keep this node in the same place. You can either move the whole subtree down or use other arrangements.

**Reply by guoranjun2** [reply]
Ok. I see, I will let it follow the layout of the parent node. it's still nice.
